app/features/settings/settings-sidebar-provider.tsx

```diff
@@ -0,0 +1,64 @@
+import { type ReactNode, createContext, useContext, useState } from 'react';
+import type {
+  NavigateToView,
+  SettingsSidebarState,
+  SettingsView,
+} from './types';
+
+interface SettingsSidebarContextType {
+  state: SettingsSidebarState;
+  navigateToView: NavigateToView;
+}
+
+interface SettingsSidebarProviderProps {
+  children: ReactNode;
+}
+
+const SettingsSidebarContext = createContext<
+  SettingsSidebarContextType | undefined
+>(undefined);
+
+const defaultState: SettingsSidebarState = {
+  view: 'main',
+};
+
+export function SettingsSidebarProvider({
+  children,
+}: SettingsSidebarProviderProps) {
+  const [state, setState] = useState<SettingsSidebarState>(defaultState);
+
+  const navigateToView: NavigateToView = (
+    view: SettingsView,
+    params?: { accountID: string },
+  ) => {
+    if (view === 'single-account') {
```

Comment:
this seems fishy. what is going on here?

Reply:
Just trying to get TS to like me. This makes it so that params.accountID is defined. How can I do things differently so that I can just directly set the state without having to check for different views